nixos/modules/services/databases/redis.nix

```diff
@@ -54,7 +54,13 @@ in {
   options = {
 
     services.redis = {
-      package = mkPackageOption pkgs "redis" { };
+      package = mkOption {
+        type = types.package;
+        default = pkgs.valkey;
+        defaultText = literalExpression "pkgs.valkey";
+        description = "Which package to use for the Redis service.";
+        relatedPackages = [ "valkey" "redis" ];
+      };
```

Comment:
Did you think about guarding this behind system.stateVersion? What's the reasoning to not do it?

I am fully sure Valkey is fully compatible but there is probably some software out there which might be not that small, which matches the product name in some case and we could break that and I would like to avoid doing that silently.